src/qmltypes/qmlfilter.cpp

```diff
@@ -458,7 +468,27 @@ void QmlFilter::preset(const QString &name)
 
     if (!dir.cd("presets") || !dir.cd(objectNameOrService()))
         return;
-    m_filter.load(dir.filePath(name).toUtf8().constData());
+
+    // Detect the preset file format
+    bool isYaml = false;
+    QFile presetFile(dir.filePath(name));
+    if(presetFile.open(QIODevice::ReadOnly)) {
+        if(presetFile.readLine(4) == "---") {
+            isYaml = true;
+        }
+        presetFile.close();
+    }
+
+    if(isYaml) {
+        // Load from YAML file.
+        Mlt::Properties* properties = Mlt::Properties::parse_yaml(dir.filePath(name).toUtf8().constData());
+        m_filter.pass_values(*properties, "");
+        delete properties;
```

Comment:
Preference now is to use smart pointers:
`QScopedPointer<Mlt::Properties> properties(Mlt::Properties::parse_yaml(dir.filePath(name).toUtf8().constData()));`

Reply:
done